x/ccv/parent/keeper/relay.go

```diff
@@ -113,7 +118,57 @@ func (k Keeper) EndBlockCallback(ctx sdk.Context) {
 		if len(valUpdates) != 0 {
 			k.SendPacket(ctx, chainID, valUpdates, valUpdateID)
 			k.IncrementValidatorSetUpdateId(ctx) // TODO: this needs to be moved out of this scope
+			k.SetValsetUpdateBlockHeight(ctx, valUpdateID, uint64(ctx.BlockHeight()))
 		}
 		return false
 	})
 }
+
+// OnRecvPacket slahes and jails the given validator in the packet data
+func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data ccv.ValidatorDowntimePacketData) exported.Acknowledgement {
+	if childChannel, ok := k.GetChannelToChain(ctx, packet.DestinationChannel); !ok && childChannel != packet.DestinationChannel {
```

Comment:
`GetChannelToChain` returns the chain ID associated to the `packet.DestinationChannel`. You shouldn't compare the result to a channel ID.